dll/cpl/desk/desktop.c

```diff
@@ -81,6 +112,47 @@ InitDesktopSettings(PDESKTOP_DATA pData)
         RegCloseKey(iconKey2);
 
     RegCloseKey(regKey);
+
+LoadIcons:
+    for (i = 0; i < _countof(IconChange); i++)
+    {
+        TCHAR iconPath[MAX_PATH];
+        TCHAR szData[MAX_PATH];
+
+        /* Current icons */
+        StringCchCopy(iconPath, _countof(iconPath), szUserClass);
+        StringCchCat(iconPath, _countof(iconPath), IconChange[i].CLSID);
+        StringCchCat(iconPath, _countof(iconPath), szDefaultIcon);
+        cbData = sizeof(szData);
+
+        if (SHGetValue(HKEY_CURRENT_USER, iconPath, IconChange[i].Icon, &dwType,
+                       &szData, &cbData) == ERROR_SUCCESS &&
+            (dwType == REG_SZ || dwType == REG_EXPAND_SZ))
+        {
+            StringCchCopy(pData->Icon[i].szPath, _countof(pData->Icon[i].szPath), szData);
+        }
+
+        /* Default icons */
+        /* FIXME: Get default icons from theme data, fallback to CLSID data on error. */
+        StringCchCopy(iconPath, _countof(iconPath), szSysClass);
+        StringCchCat(iconPath, _countof(iconPath), IconChange[i].CLSID);
+        StringCchCat(iconPath, _countof(iconPath), szDefaultIcon);
+        cbData = sizeof(szData);
+
+        if (SHGetValue(HKEY_CLASSES_ROOT, iconPath, IconChange[i].Icon, &dwType,
+                       &szData, &cbData) == ERROR_SUCCESS &&
+            (dwType == REG_SZ || dwType == REG_EXPAND_SZ))
+        {
+            StringCchCopy(pData->DefIcon[i].szPath, _countof(pData->DefIcon[i].szPath), szData);
+        }
+
+        /* Emergency fallback */
+        if (lstrlen(pData->DefIcon[i].szPath) == 0)
```

Comment:
```suggestion
        if (!*(pData->DefIcon[i].szPath))
```

Reply:
`szPath` is an array, not a pointer to string. I think `if (!pData->DefIcon[i].szPath[0])` would be more correct in this case, however currently it's more readable and easy to understand. I prefer to leave it as is.

Reply:
> szPath is an array, not a pointer to string.

In C these are the same thing.

> I think `if (!pData->DefIcon[i].szPath[0])` would be more correct [...]

It is exactly the same, and maybe more readable.